cpp/src/lance/io/reader.cc

```diff
@@ -49,26 +49,61 @@ typedef ::arrow::Result<std::shared_ptr<::arrow::Scalar>> ScalarResult;
 
 namespace lance::io {
 
+namespace {
+
 ::arrow::Result<int64_t> ReadFooter(const std::shared_ptr<::arrow::Buffer>& buf) {
   assert(buf->size() >= 16);
   if (auto magic_buf = ::arrow::SliceBuffer(buf, buf->size() - 4);
       !magic_buf->Equals(::arrow::Buffer(lance::format::kMagic))) {
     return Status::IOError(
         fmt::format("Invalidate file format: MAGIC NUM is not {}", lance::format::kMagic));
   }
-  // Metadata Offset
   return ReadInt<int64_t>(buf->data() + buf->size() - 16);
 }
 
+}  // namespace
+
 ::arrow::Result<std::unique_ptr<FileReader>> FileReader::Make(
     std::shared_ptr<::arrow::io::RandomAccessFile> in,
     std::shared_ptr<::lance::format::Manifest> manifest,
     ::arrow::MemoryPool* pool) {
-  auto reader = std::make_unique<FileReader>(in, manifest, pool);
+  auto reader = std::make_unique<FileReader>(std::move(in), std::move(manifest), pool);
   ARROW_RETURN_NOT_OK(reader->Open());
   return reader;
 }
 
+::arrow::Result<std::shared_ptr<::lance::format::Manifest>> FileReader::OpenManifest(
+    const std::shared_ptr<::arrow::io::RandomAccessFile>& in) {
+  constexpr auto kBufReadBytes = 8 * 1024 * 1024;  // Read 8 MB;
```

Comment:
why 8 MB?

Reply:
This is just an arbitrarily `not-to-small` number.  Setting this too small, it could result to more than one IO to read manifest file. 

Has a length here was due to arrow `RandomAccessFile` does not the function of "real the full file" API. It can be larger tho.